php/libraries/DefaultFilesUploadHandler.class.inc

```diff
@@ -0,0 +1,367 @@
+<?php
+/**
+ * Default class for handling file uploads in LORIS
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * Provides a default handler for files that were uploaded on a LORIS
+ * web site using LorisForm.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+class DefaultFilesUploadHandler implements FilesUploadHandler
+{
+    /**
+     * Default directory to move the uploaded files to if none is explicitely
+     * specified.
+     */
+    public static $DEFAULT_MOVE_DIR = __DIR__ . "../user_uploads";
+    /**
+     *  The LorisForm used to upload the file(s).
+     */
+    private $_lorisForm;
+    /**
+     * Directory to move the uploaded files to. Will be created if it does not
+     * exist.
+     */
+    private $_moveDirPath;
+    /**
+     * When creating $_moveDirPath, these are the access rights to give it
+     * (should be an octal number).
+     */
+    private $_dirPermissions;
+    /**
+     * When creating $_moveDirPath, give it this access group.
+     */
+    private $_dirGroup;
+    /**
+     * All the errors that occured either when the files are validated, moved to
+     * their final locations or processed.
+     */
+    private $_errors;
+    /**
+     * The files and their final locations (if successfully moved). Each value
+     * in this array is a SplFileInfo. Each key is the name of the LorisForm
+     * field used to upload the file.
+     */
+    private $_movedFiles;
```

Comment:
This (and most of the things that have higher than private getter/setters) should be protected.